libs/langchain/langchain/vectorstores/redis/redis.py

```diff
@@ -281,11 +316,47 @@ def similarity_search(
         Returns:
             List[Document]: A list of documents that are most similar to the query text.
         """
-        docs_and_scores = self.similarity_search_with_score(query, k=k)
-        return [doc for doc, _ in docs_and_scores]
+
+        # Creates embedding vector from user query
+        embedding = self.embedding_function(query)
+
+        # Creates Redis query
+        params_dict: Dict[str, Union[str, bytes]] = {
+            "vector": array_to_buffer(embedding),
+        }
+
+        if score_threshold is None:
+            redis_query = self._prepare_vector_query(k, filter=filter)
+        else:
+            redis_query = self._prepare_range_query(k, filter=filter)
```

Comment:
is redis_query supposed to be identical in both if branches?

Reply:
yes.